include/zephyr/bluetooth/mesh/access.h

```diff
@@ -703,16 +712,14 @@ struct bt_mesh_model {
 	/** Model callback structure. */
 	const struct bt_mesh_model_cb * const cb;
 
-#ifdef CONFIG_BT_MESH_MODEL_EXTENSIONS
-	/* Pointer to the next model in a model extension list. */
-	struct bt_mesh_model *next;
-#endif
-
 #ifdef CONFIG_BT_MESH_LARGE_COMP_DATA_SRV
 	/* Pointer to the array of model metadata entries. */
 	struct bt_mesh_models_metadata_entry **metadata;
 #endif
 
+	/** Model ctx variables. */
+	struct bt_mesh_model_ctx * const data;
```

Comment:
Generally the changes in this PR look fine to me, but could you consider renaming this variable to `ctx` instead of `data`? I think that would be slightly more descriptive than `data`. I guess the expanded version of this would be something like "Runtime Context", i.e. information that isn't known at build-time, but `ctx` seems like a good enough shorthand for that.